### DISCUSSION TITLE
[MongoDb] Can I Update an field of an item of an array in my file

### DISCUSSION THREAD

**Discussion by Dsantosjaime** [prisma/prisma#20378] [Q&A]
I have this Schema:

```prisma
model User {
  client     Client?
}

type Client {
  sites SiteInfos[]
}

type SiteInfos {
  id   String @db.ObjectId
  name String
}
```

How can I update all User.client.sites[x].name by "newName" where site.id === 0 ?

Actually I have this:

```js
this.prisma.user.updateMany({
                where: {
                    userType: UserType.CLIENT,
                    client: {
                        is: {
                            sites: {
                                some: {
                                    id
                                }
                            }
                        }
                    }
                },
                data: {}
            }),
```

But i can't find out the "data" part.
I think I should use "aggregate" ?

**Comment by nurul3101** [comment]
Hello @Dsantosjaime 👋 

One easier/straightforward approach I can think of is that you can get the users' sites and update it in your application code. Like this:

```ts
  const user = await prisma.user.findUnique({
    where: { id: userId },
    include: { client: true }
  });

  // Update the site name
  user.client.sites = user.client.sites.map(site =>
    site.id === siteId ? { ...site, name: newName } : site
  );

  // Save the user back to the database
  await prisma.user.update({
    where: { id: userId },
    data: { client: user.client }
  });
```

Would this approach work for you?

**Reply by nurul3101** [reply]
Do let us know in case you still need assistance.

If not, please consider marking the discussion as answered/closed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by chouzzy** [comment]
Hi guys, is this the only way to update it? There's no way to search the id on the update query?

feels like something like this should exist:

```
const investmentUpdated = await prisma.investment.update({
            where: { id: userInvestmentFound.id },
            data: {
                apartaments: {
                    updateMany: {
                        where: {
                            id: userInvestmentFound.apartament?.id,
                        },
                        data: { userId: null }
                    }
                }
            }
        })
```
